tests/test_tecan_od_analyzer.py

```diff
@@ -0,0 +1,208 @@
+ 		result1 = time_formater(df_gr)
+ 		self.assertEqual(len(result1["Sample_ID"]),len(df_gr["Sample_ID"]))
+
+
+
+	#----- TEST volume_loss_correlation METHOD -----
+
+
+ 	def test_volume_loss_correlation_returned_variable(self):
+ 		"""test that the returned variables are dataframes with correlation column containing floats"""
+ 		result1 = vol_correlation(df_vl_time)
+ 		colnames1 = (result1.columns.values)
+ 		self.assertIsInstance(result1, pd.DataFrame)
+ 		self.assertIn("Correlation", colnames1)
+ 		self.assertTrue(ptypes.is_float_dtype(result1["Correlation"]))
+
+
+ 	def test_volume_loss_correlation_correctness(self):
+ 		"""test that the correlation is computed by dividing the first OD time point on every other value belonging to the same bioshaker"""
+ 		result1 = vol_correlation(df_vl_time)
+ 		
+ 		#Assess manually the row containing the first OD time point and bioshaker
+ 		t0_value_row = df_vl_time.iloc[0]
+ 		t0_sample = t0_value_row["Sample_ID"]
+
+ 		#Subset the df_time to 20 rows to compute the correlation with less data and we make sure they only contain the same bioshaker as in t0
+ 		df_vl_temp = df_vl_time[df_vl_time["Sample_ID"] == t0_sample]
+ 		
+ 		#Compute the correlation manually and with the method we are testing
+ 		man_result = df_vl_temp["Measurement"] / t0_value_row["Measurement"]
+ 		method_result = vol_correlation(df_vl_time)
+ 		method_result = method_result[method_result["Sample_ID"] == t0_sample]
+ 		method_result = method_result["Correlation"]
+ 		
+ 		#Assert that both correlations are equal
+ 		self.assertEqual(man_result.tolist(), method_result.tolist())
+		
+ 		
+
+	#----- TEST compensation_lm METHOD -----
+
+
+ 	def test_compensation_lm(self):
+ 		"""test the compensation_lm method"""
+
+ 		result1, result2 = compensation_lm(cor_df, df_gr_time)
+ 		colnames = (result2.columns.values)
+ 		self.assertIsNotNone(result1)
+ 		self.assertIsInstance(result2, pd.DataFrame)
+ 		self.assertIn("Corrected_Measurement", colnames)
+ 		self.assertTrue(ptypes.is_float_dtype(result2["Corrected_Measurement"]))
+
+
+ 	def test_compensation_lm_check_not_loosen_samples(self):
+ 		"""test the compensation_lm method"""
+ 		result1, result2 = compensation_lm(cor_df, df_gr_time)
+ 		self.assertEqual(len(result2), len(df_gr_time))
+
+
+
+ 	#----- TEST reshape_gr METHOD -----
+
+
+ 	def test_reshape_dataframe(self):
+ 		"""test the reshape method, checks that output is a dataframe and that all the Sample_IDs are contained"""
+ 		result = reshape_dataframe(df_gr_comp)
+ 		original_IDs = df_gr_comp["Sample_ID"].unique()
+ 		colnames = (result.columns.values)
+ 		self.assertIsInstance(result, pd.DataFrame)
+ 		for name in colnames :
+ 			name = re.search(r"([B][S]\d[.]\S*[_]*\S*)", name).group(1)
+ 			self.assertIn(str(name), str(original_IDs))
+
+ 	###IMPLEMENT THE TEST OF THE FLAG
+
+
+
+ 	#----- TEST gr_estimation METHOD -----
+
+ 	
+ 	def test_gr_estimation(self):
```

Comment:
I can see that some of the tests now are looking at the values generated by running the different methods.  Please check that all methods that change the dataframe values have some test in place that checks the correctness of the values.  For example, it looks like there should be something here that tests the different values.